src/installer.rs

```diff
@@ -60,91 +63,135 @@ impl Display for LoaderVersion {
 }
 
 pub async fn fetch_minecraft_versions() -> Result<Vec<MinecraftVersion>> {
-    Ok(reqwest::get("https://meta.quiltmc.org/v3/versions/game").await?.json().await?)
+    Ok(get("https://meta.quiltmc.org/v3/versions/game")
+        .await?
+        .json()
+        .await?)
 }
 
 pub async fn fetch_loader_versions() -> Result<Vec<LoaderVersion>> {
-    Ok(reqwest::get("https://meta.quiltmc.org/v3/versions/loader").await?.json().await?)
+    Ok(get("https://meta.quiltmc.org/v3/versions/loader")
+        .await?
+        .json()
+        .await?)
 }
 
-#[derive(Serialize, Deserialize)]
-struct LaunchProfiles {
-    profiles: HashMap<String, serde_json::Value>,
-    settings: serde_json::Value,
-    version: u32
+/// `Deserialize` is not implemented for a reason
+///
+/// DO NOT deserialise `launcher_profiles.json` into this incomplete struct and write it back as it will cause **data loss**
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Profile {
+    name: String,
+    #[serde(rename = "type")]
+    profile_type: String,
+    created: DateTime<Utc>,
+    last_version_id: String,
+    icon: String,
 }
 
 pub async fn install_client(args: ClientInstallation) -> Result<()> {
     println!("Installing client: {:#?}", args);
 
     // Verify install location
     if !args.install_location.exists() {
-        return Err(anyhow!("Target directory doesn't exist: {:?}", args.install_location));
+        return Err(anyhow!(
+            "Target directory doesn't exist: {:?}",
+            args.install_location
+        ));
```

Comment:
One line might be better here

Reply:
This reminds me, when displaying paths to the end-user, `path.display()` is preferred. (https://stackoverflow.com/questions/57944423/should-i-prefer-path-display-over-debug-formatting)
This would probably be better on multiple lines, what do you think?

Reply:
Hm, try single-line first; I still feel like multi-line would be too much here, but yeah, I'll wanna see it in practice first in order to make that judgement :P

Reply:
In general for an error, I think debug is fine.
But this would likely be often user-facing, so display makes more sense.
(And then it's longer so Ennui will accept multi-line)

Reply:
> (And then it's longer so Ennui will accept multi-line)

@EnnuiL [I've already made the change](https://github.com/QuiltMC/quilt-native-installer/pull/13/files#diff-4904922dab3fd7bd66e213f49476ad3aa39a8a7c0cb24f68a68343909bbdaab4R93-R96), let me know what you think

Reply:
Yep, maybe multi-line would be better here